src/msg/msg_types.h

```diff
@@ -384,6 +385,31 @@ struct entity_addr_t {
     }
   }
 
+  std::string ip_only_to_str() {
+    const char *host_ip = NULL;
+    char addr_buf[INET6_ADDRSTRLEN];
+    std::string ip_str;
+    switch (u.sa.sa_family) {
+    case AF_INET:
+      host_ip = inet_ntop(AF_INET, &u.sin.sin_addr, 
+                          addr_buf, INET_ADDRSTRLEN);
+      break;
+    case AF_INET6:
+      host_ip = inet_ntop(AF_INET6, &u.sin6.sin6_addr, 
+                          addr_buf, INET6_ADDRSTRLEN);
+      break;
+    default:
+      break;
+    }
+
+    if (NULL == host_ip) {
+      ip_str = "";
+    } else {
+      ip_str = host_ip; 
+    }
+    return ip_str;
```

Comment:
this can be simplified to <pre>
return host_ip ? host_ip : "";
</pre>
and drop the ip_str decl

Otherwise, looks good, thanks!

Reply:
@liewegas 
“Less is More”  :-)
updated it